packages/pricing/src/services/pricing-module.ts

```diff
@@ -1750,11 +1775,49 @@ export default class PricingModuleService<
     data: PricingTypes.AddPriceListPricesDTO[],
     sharedContext: Context = {}
   ): Promise<PricingTypes.PriceListDTO[]> {
+    const ruleTypeAttributes: string[] = []
+    const priceListIds: string[] = []
+
+    for (const priceListData of data) {
+      priceListIds.push(priceListData.priceListId)
+
+      for (const price of priceListData.prices) {
+        ruleTypeAttributes.push(...Object.keys(price.rules || {}))
+      }
+    }
+
+    const ruleTypes = await this.listRuleTypes(
+      { rule_attribute: ruleTypeAttributes },
+      { take: null }
+    )
+
+    const ruleTypeMap: Map<string, RuleTypeDTO> = new Map(
+      ruleTypes.map((rt) => [rt.rule_attribute, rt])
+    )
+
+    for (const ruleAttribute of arrayUnique<string>(ruleTypeAttributes)) {
+      let ruleType = ruleTypeMap.get(ruleAttribute)
+
+      if (ruleType) {
+        continue
+      }
+
+      ;[ruleType] = await this.createRuleTypes(
+        [
+          {
+            name: ruleAttribute,
+            rule_attribute: ruleAttribute,
+          },
+        ],
+        sharedContext
+      )
+
+      ruleTypeMap.set(ruleAttribute, ruleType)
+    }
```

Comment:
We generally don't allow new rule-types to be created if the rule key is passed along. When adding price-set prices we are validating that rule-types already exists, if not, we raise an exception. We should keep this behavior here